controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -239,7 +239,36 @@ bool master_thread::handle_cmdu_1905_autoconfiguration_search(Socket *sd,
 
     LOG(DEBUG) << "Received AP_AUTOCONFIGURATION_SEARCH_MESSAGE";
 
-    auto tlvAlMacAddressType = cmdu_rx.addClass<ieee1905_1::tlvAlMacAddressType>();
+    std::shared_ptr<ieee1905_1::tlvAlMacAddressType> tlvAlMacAddressType     = nullptr;
+    std::shared_ptr<ieee1905_1::tlvSearchedRole> tlvSearchedRole             = nullptr;
+    std::shared_ptr<ieee1905_1::tlvAutoconfigFreqBand> tlvAutoconfigFreqBand = nullptr;
+    std::shared_ptr<wfa_map::tlvSupportedService> tlvSupportedServiceIn      = nullptr;
+    std::shared_ptr<wfa_map::tlvSearchedService> tlvSearchedService          = nullptr;
+    int type = cmdu_rx.getNextTlvType();
+
+    while ((type = cmdu_rx.getNextTlvType()) != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE)) {
+        if (type == int(wfa_map::eTlvTypeMap::TLV_SUPPORTED_SERVICE)) {
+            LOG(DEBUG) << "Found TLV_SUPPORTED_SERVICE TLV";
+            tlvSupportedServiceIn = cmdu_rx.addClass<wfa_map::tlvSupportedService>();
+        } else if (type == int(ieee1905_1::eTlvType::TLV_AL_MAC_ADDRESS_TYPE)) {
+            LOG(DEBUG) << "Found TLV_AL_MAC_ADDRESS_TYPE TLV";
+            tlvAlMacAddressType = cmdu_rx.addClass<ieee1905_1::tlvAlMacAddressType>();
+        } else if (type == int(ieee1905_1::eTlvType::TLV_SEARCHED_ROLE)) {
+            LOG(DEBUG) << "Found TLV_SEARCHED_ROLE TLV";
+            tlvSearchedRole = cmdu_rx.addClass<ieee1905_1::tlvSearchedRole>();
+        } else if (type == int(ieee1905_1::eTlvType::TLV_AUTOCONFIG_FREQ_BAND)) {
+            LOG(DEBUG) << "Found TLV_AUTOCONFIG_FREQ_BAND TLV";
+            tlvAutoconfigFreqBand = cmdu_rx.addClass<ieee1905_1::tlvAutoconfigFreqBand>();
+        } else if (type == int(wfa_map::eTlvTypeMap::TLV_SEARCHED_SERVICE)) {
+            LOG(DEBUG) << "Found TLV_SEARCHED_SERVICE TLV";
+            tlvSearchedService = cmdu_rx.addClass<wfa_map::tlvSearchedService>();
+        } else {
+            LOG(ERROR) << "Unknown TLV, type " << std::hex << type;
```

Comment:
Please update the error log to indicate that this autoconfig search message is **dropped**
```suggestion
            LOG(ERROR) << "Unknown TLV, type " << std::hex << type << ", dropping...";
```